### PR TITLE
docs: how to add a model

### PR DESCRIPTION
**Pull request by phymbert**
### Motivation

As I suffered a little in:
- #6515 

I hope it can be useful for the next contributions.

### CODE REVIEW

File: docs/HOWTO-add-model.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+Have a look to existing implementation like `build_llama`, `build_dbrx` or `build_bert`.
+
+When implementing a new graph, please note that the underlying `ggml` backends do not support them all, support of missing backend operations can be added in another PR.
+
+## Terminology
+
+| term    | description                                                                                                                                       | link                                                       |
+|---------|---------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------|
+| GGML    | Georgi Gerganov Model Language a.k.a GPT-Generated Model Language                                                                                 | https://github.com/ggerganov/ggml                          |
```
Comment (Green-Sky) [inline docs/HOWTO-add-model.md:114]: this sounds wrong, what "ggml" does this refer to? was this ever a thing?

Comment (phymbert) [reply docs/HOWTO-add-model.md:114]: Which one ? I feel both sound correct, I heard the first time about GGML here:
https://medium.com/@phillipgimmi/what-is-gguf-and-ggml-e364834d241c

@ggerganov I feel funny to have both `Georgi Gerganov Model Language` and `GPT-Generated Model Language` but nothing is stated on the project page, maybe this is a big secret :D Up to you

**Comment by maddes8cht** [comment]
I always belived it means Georgi Gerganov machine learning...

**Comment by phymbert** [comment]
> I always belived it means Georgi Gerganov machine learning...

We will never know :P

File: README.md [hunk 2]
```diff
@@ -122,6 +122,8 @@ Typically finetunes of the base models below are supported as well.
 - [x] [SEA-LION](https://huggingface.co/models?search=sea-lion)
 - [x] [GritLM-7B](https://huggingface.co/GritLM/GritLM-7B) + [GritLM-8x7B](https://huggingface.co/GritLM/GritLM-8x7B)
 
+Your model is not yet in the list ? New architecture contributions are welcomed, have a look to [Add a model.md](./docs/HOWTO-add-model.md) HOW TO.
```
Comment (ggerganov) [inline README.md:125]:
```suggestion
(instructions for supporting more models: [HOWTO-add-model.md](./docs/HOWTO-add-model.md))
```

**Review by ggerganov — APPROVED** [review]

**Pull request merged**